public/app/plugins/datasource/cloudwatch/utils/logsRetry.ts

```diff
@@ -39,83 +39,56 @@ export function runWithRetry(
     // Run function is where the logic takes place. We have it in a function so we can call it recursively.
     function run(currentQueryParams: StartQueryRequest[]) {
       subscription = queryFun(currentQueryParams).subscribe({
-        next(frames) {
-          // In case we successfully finished, merge the current response with whatever we already collected.
-          const collectedPreviously = toDataQueryResponse({ data: { results: collected } }).data || [];
-          observer.next({ frames: [...collectedPreviously, ...frames] });
-          observer.complete();
-        },
-        error(error: FetchError<{ results?: Record<string, Result> }> | string) {
-          // In case of error we first try to figure out what kind of error it is
-
-          // This means it was a generic 500 error probably so we just pass it on
-          if (typeof error === 'string') {
-            observer.error(error);
-            return;
-          }
-
-          // In case of multiple queries this some can error while some may be ok
-          const errorData = splitErrorData(error);
+        next(response: DataQueryResponse) {
+          if (response.errors) {
+            const { refIdsForRequestsToRetry, errorsNotToRetry } = splitErrorsData(response.errors);
+            if (refIdsForRequestsToRetry.length > 0) {
+              if (!timeoutFunc(retries, startTime.valueOf())) {
+                // store the responses we are not retrying
+                collected.data = [
+                  ...collected.data,
+                  ...response.data,
```

Comment:
does `...response.data` include every response? E.g. if a response has an error would that be included here even if it was an error response that would be retried?

Reply:
I think if it had an error it would be in response.errors no? 

When I call this with 2 queries (a and b) and log it out, runWithRetry is called separately for each query, I'm not sure I follow in what circumstances response.data would have more than one entry in but it does come back as an array.